tests/test_run_build.py

```diff
@@ -160,3 +161,45 @@ def test_main_manifest_new_lock_with_overrides(self, mock_logging: Mock, mock_to
         mock_stable.assert_called_with()
         mock_to_file.assert_called_with(self.OPENSEARCH_MANIFEST + ".lock")
         mock_logging.assert_called_with(f"Creating {self.OPENSEARCH_MANIFEST}.lock")
+
+    @patch("argparse._sys.argv", ["run_build.py", OPENSEARCH_MANIFEST, "-p", "linux", "--continue-on-error"])
+    @patch("run_build.Builders.builder_from", return_value=MagicMock())
+    @patch("run_build.BuildRecorder", return_value=MagicMock())
+    @patch("run_build.TemporaryDirectory")
+    @patch("run_build.logging.error")
+    def test_fail_core_component_continue_on_error(self, mock_logging_error: Mock, mock_temp: Mock, mock_recorder: Mock, mock_builder_from: Mock, *mocks: Any) -> None:
+        mock_temp.return_value.__enter__.return_value.name = tempfile.gettempdir()
+        mock_builder = Mock()
+        mock_builder.build.side_effect = Exception("Error during build")
+        mock_builder_from.return_value = mock_builder
+        with pytest.raises(Exception, match="Error during build"):
+            main()
+        mock_logging_error.assert_called_with(f"Error building OpenSearch, retry with: run_build.py {self.OPENSEARCH_MANIFEST} --component OpenSearch")
+
+    @patch("argparse._sys.argv", ["run_build.py", NON_OPENSEARCH_MANIFEST, "-p", "linux", "--continue-on-error"])
+    @patch("run_build.Builders.builder_from", return_value=MagicMock())
+    @patch("run_build.BuildRecorder", return_value=MagicMock())
+    @patch("run_build.TemporaryDirectory")
+    @patch("run_build.logging.error")
+    def test_fail_plugins_continue_on_error(self, mock_logging_error: Mock, mock_temp: Mock, mock_recorder: Mock, mock_builder_from: Mock, *mocks: Any) -> None:
+        mock_temp.return_value.__enter__.return_value.name = tempfile.gettempdir()
+        mock_builder = Mock()
+        mock_builder.build.side_effect = Exception("Error during build")
+        mock_builder_from.return_value = mock_builder
+
+        main()
+        mock_logging_error.assert_called_with("Failed plugins are ['common-utils', 'job-scheduler', 'sql', 'alerting']")
```

Comment:
So this is failing but OpenSearch is not a part of the manifest?

Reply:
Yes, the test logic picks the first component from the manifest and mocks an exception, had to remove OS to test behavior is the component is a plugin.

Reply:
But in case of core failure we want the entire build to fail right?

Reply:
Yes `def test_fail_core_component_continue_on_error` tests exception is raised, not just caught.